### PR TITLE
feat: add iterator syntax

### PR DESCRIPTION
**Pull request by randi274**
### What does this PR do?
Added the iterator:it template directive to suggested attributes with others like for:each and
if:true. Fixed issue where description wasn't showing on any of the directives, and added links to
our docs as well. This also allow for flexible syntax, where the user wants to use another variable 
for 'it'.

### What issues does this PR fix or reference?
@W-10536569@, @W-10542295@

### What was the behavior before?
The template directive iterator:it was not appearing in our list of auto-complete attributes, and description information was broken.

### What was the behavior now?
The directive appears, and the description information is now available for all other template directives. 🎉

### CODE REVIEW

File: packages/lwc-language-server/scripts/convert.js [hunk 1]
```diff
@@ -50,26 +50,71 @@ const tags = Object.keys(data).map(key => {
     };
 });
 
+//make globalAttribute changes here, not in standard-lwc.json as they'll be overwritten
```
Comment (jeffb-sfdc) [inline packages/lwc-language-server/scripts/convert.js:53]: OK, this is super nin-picky, but I thought out ES-lint rules required a space after the `//`

Comment (jeffb-sfdc) [reply packages/lwc-language-server/scripts/convert.js:53]: Other than this, the PR looks good.

File: packages/lwc-language-server/src/resources/standard-lwc.json [hunk 2]
```diff
@@ -3649,23 +3649,66 @@
   "globalAttributes": [
     {
       "name": "for:each",
-      "decription": "Renders the element or template block multiple times based on the expression value."
+      "description": "Renders the element or template block multiple times based on the expression value.",
```
Comment (gbockus-sf) [inline packages/lwc-language-server/src/resources/standard-lwc.json:3652]: this typo found it's way everywhere :)

File: packages/lwc-language-server/scripts/convert.js [hunk 3]
```diff
@@ -50,26 +50,71 @@ const tags = Object.keys(data).map(key => {
...
 const globalAttributes = [
```
Comment (gbockus-sf) [inline packages/lwc-language-server/scripts/convert.js:54]: nit: so what way we could have caught his is if 1. this was a ts file and 2. we defined a type/interface for the attributes to implement.

File: packages/lwc-language-server/scripts/convert.js [hunk 4]
```diff
@@ -50,26 +50,71 @@ const tags = Object.keys(data).map(key => {
...
+            kind: 'markdown',
+            value:
+                'Bind the current iteration item to an identifier. Contains properties (`value`, `index`, `first`, `last`) that let you apply special behaviors to certain items.',
+        },
+        references: [
+            {
+                name: 'Salesforce',
+                url: 'https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.reference_directives',
+            },
+        ],
```
Comment (gbockus-sf) [inline packages/lwc-language-server/scripts/convert.js:117]: dumb q: are these dups for what's in [standard-lwc.json](https://github.com/forcedotcom/lightning-language-server/pull/488/files#diff-e30348f58226e23daf10a57ff87c0b636112f69a5de89aa0434c6344403d961d). If so what don't we just import that here and use it?

File: packages/lwc-language-server/scripts/convert.js (hunk 3 above)
Comment (gbockus-sf) [reply packages/lwc-language-server/scripts/convert.js:54]: this is more just an observation.  no action to be taken here.

**Review by gbockus-sf — APPROVED** [review]
changes seem good.  One suggestion to consider if the json data and the js data are intended to be the same.

File: packages/lwc-language-server/scripts/convert.js (hunk 1 above)
Comment (randi274) [reply packages/lwc-language-server/scripts/convert.js:53]: I think something's amiss with eslint here; @gbockus-sf gave similar feedback on a previous PR in here! I will get that done in short order though 👍

File: packages/lwc-language-server/src/resources/standard-lwc.json (hunk 2 above)
Comment (randi274) [reply packages/lwc-language-server/src/resources/standard-lwc.json:3652]: It really did 😆

File: packages/lwc-language-server/scripts/convert.js (hunk 3 above)
Comment (randi274) [reply packages/lwc-language-server/scripts/convert.js:54]: 💯

File: packages/lwc-language-server/scripts/convert.js (hunk 4 above)
Comment (randi274) [reply packages/lwc-language-server/scripts/convert.js:117]: So we have a script that runs every time we build that calls `convert.js`, and that is what then populates `standard-lwc.json`. I was trying to dig up if there was still a purpose to having this separately, but wasn't able to find a clear answer in any conversation/PR/commit message. That said, globalAttributes goes back a long time in this file, and I do know for sure that this overwrites whatever we put in to standard-lwc.

Comment (gbockus-sf) [reply packages/lwc-language-server/scripts/convert.js:117]: gotcha

**Review by jeffb-sfdc — APPROVED** [review]

**Pull request merged**